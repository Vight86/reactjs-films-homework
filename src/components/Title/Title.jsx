import React from 'react';
import propTypes from 'prop-types';
import styles from './Title.scss';

const Title = ({ className, children }) => {
  switch (className) {
    case 'secondary':
    case 'tertiary':
      return (
        <h3 className={styles[className]}>
          { children}
        </h3>
      );
    default:
      return (
        <h1 className={styles[className]}>
          {children}
        </h1>
      );
  }
};

Title.propTypes = {
  className: propTypes.string.isRequired,
  children: propTypes.oneOfType([
    propTypes.element,
    propTypes.string,
  ]).isRequired,
};

export default Title;
