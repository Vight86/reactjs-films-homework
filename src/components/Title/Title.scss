@import '../../utils/variables';

.title {
  overflow: hidden;
  font-weight: 700;
  letter-spacing: 0.04em;
  line-height: normal;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}

.primary {
  @extend .title;

  margin: 0 0 0.5em 0;
  color: $color-light;
  font-size: 6em;
}

.secondary {
  @extend .title;

  margin: 0 0 0.71em 0;
  color: $text-tertiary;
  font-size: 2.8em;
}

.tertiary {
  @extend .secondary;

  color: $color-light;
}
