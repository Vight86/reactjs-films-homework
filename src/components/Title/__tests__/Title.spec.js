import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Title from '../index';

const setUp = (className, title) => {
  const renderer = new ShallowRenderer();
  renderer.render(<Title className={className}>{title}</Title>);
  return renderer.getRenderOutput();
};

describe('render Title component', () => {
  it('render without props', () => {
    const result = setUp();
    expect(result).toMatchSnapshot();
  });

  it('render className secondary correctly', () => {
    const result = setUp('secondary', 'title');
    expect(result).toMatchSnapshot();
  });

  it('render className tertiary correctly', () => {
    const result = setUp('tertiary', 'title');
    expect(result).toMatchSnapshot();
  });
});
