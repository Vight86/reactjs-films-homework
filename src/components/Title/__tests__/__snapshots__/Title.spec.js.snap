// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Title component render className secondary correctly 1`] = `
<h3
  className="secondary"
>
  title
</h3>
`;

exports[`render Title component render className tertiary correctly 1`] = `
<h3
  className="tertiary"
>
  title
</h3>
`;

exports[`render Title component render without props 1`] = `
<h1
  className=""
>
  
</h1>
`;
